src/NextjsDistribution.ts

```diff
@@ -459,69 +446,32 @@ export class NextjsDistribution extends Construct {
       domainNames,
       certificate: this.certificate,
-      defaultBehavior: {
-        origin: fallbackOriginGroup, // try S3 first, then lambda
-        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
-        compress: true,
-        // what goes here? static or lambda?
-        cachePolicy: lambdaCachePolicy,
-        originRequestPolicy: fallbackOriginRequestPolicy,
-        edgeLambdas: lambdaOriginEdgeFns,
-      },
+      defaultBehavior: this.serverBehaviorOptions,
 
       additionalBehaviors: {
         // is index.html static or dynamic?
-        ...(hasIndexHtml ? {} : { '/': lambdaBehavior }),
+        ...(hasIndexHtml ? {} : { '/': this.serverBehaviorOptions }),
 
         // known dynamic routes
-        'api/*': lambdaBehavior,
-        '_next/data/*': lambdaBehavior,
+        'api/*': this.serverBehaviorOptions,
+        '_next/data/*': this.serverBehaviorOptions,
 
         // dynamic images go to lambda
-        '_next/image*': imageBehavior,
-
-        // known static routes
-        // it would be nice to create routes for all the static files we know of
-        // but we run into the limit of CacheBehaviors per distribution
-        '_next/*': staticBehavior,
+        '_next/image*': this.imageBehaviorOptions,
       },
     });
+    return distribution;
   }
 
-  private createCloudFrontStaticCachePolicy(): cloudfront.CachePolicy {
-    return new cloudfront.CachePolicy(this, 'StaticsCache', NextjsDistribution.staticCachePolicyProps);
-  }
-
-  private createCloudFrontImageCachePolicy(): cloudfront.CachePolicy {
-    return new cloudfront.CachePolicy(this, 'ImageCache', NextjsDistribution.imageCachePolicyProps);
-  }
-
-  private createLambdaOriginRequestPolicy(): cloudfront.OriginRequestPolicy {
-    return new cloudfront.OriginRequestPolicy(
-      this,
-      'LambdaOriginPolicy',
-      NextjsDistribution.lambdaOriginRequestPolicyProps
-    );
-  }
-
-  private createFallbackOriginRequestPolicy(): cloudfront.OriginRequestPolicy {
-    return new cloudfront.OriginRequestPolicy(
-      this,
-      'FallbackOriginRequestPolicy',
-      NextjsDistribution.fallbackOriginRequestPolicyProps
-    );
-  }
-
-  private createImageOptimizationOriginRequestPolicy(): cloudfront.OriginRequestPolicy {
-    return new cloudfront.OriginRequestPolicy(
-      this,
-      'ImageOptPolicy',
-      NextjsDistribution.imageOptimizationOriginRequestPolicyProps
-    );
-  }
-
-  private createCloudFrontLambdaCachePolicy(): cloudfront.CachePolicy {
-    return new cloudfront.CachePolicy(this, 'LambdaCache', NextjsDistribution.lambdaCachePolicyProps);
+  private addStaticBehaviorsToDistribution() {
+    const publicFiles = fs.readdirSync(path.join(this.props.nextjsPath, '.open-next/assets'), { withFileTypes: true });
+    for (const publicFile of publicFiles) {
+      this.distribution.addBehavior(
+        publicFile.isDirectory() ? `${publicFile.name}/*` : publicFile.name,
```

Comment:
I think some escaping needs to happen here (yes it's broken in NextjsSite)
Try having a file with a `(` in the name for example
Maybe squawk with a helpful warning if behaviors is >20?

Reply:
Great callout. I will add that validation and reference documentation.